### PR TITLE
font: Implement 'float' scale text rendering

### PR DESCRIPTION
**Pull request by hajimehoshi**
'Float' scale text rendering is necessary for the shop popup.

Now ui.Credits uses font.DrawTextLang so that the original
character plane is no longer required.

This change also fixes the algorithm of (*Credits).Draw for
efficient rendering.

This change slightly changes the lineheight of the credits
to use the 'common' lineheight defined at the font package.

### CODE REVIEW

File: internal/font/font.go [hunk 1]
```diff
@@ -63,14 +64,49 @@ func init() {
...
+}
+
+var textOffscreen *ebiten.Image
+
+func DrawTextLang(screen *ebiten.Image, str string, ox, oy int, scale float64, textAlign data.TextAlign, color color.Color, displayTextRuneCount int, lang language.Tag) {
+	if isInteger(scale) {
+		drawTextLangIntScale(screen, str, ox, oy, int(scale), textAlign, color, displayTextRuneCount, lang)
+		return
+	}
+
```
Comment (wasedaigo) [inline internal/font/font.go]: so the code block after this is drawTextLangFloatScale? How about extract this into a method?

Comment (hajimehoshi) [reply internal/font/font.go]: I'm fine to break down, but would it really improve the code very much?

File: internal/font/font.go [hunk 2]
```diff
@@ -63,14 +64,49 @@ func init() {
...
+	is := int(math.Ceil(scale))
+	w, h := MeasureSize(str)
+	w *= is
+	h *= is
+	if textOffscreen != nil {
+		if tw, th := textOffscreen.Size(); tw < w || th < h {
+			textOffscreen.Dispose()
```
Comment (wasedaigo) [inline internal/font/font.go]: Is that necessary to Dispose this? I am worried about the performance if it requires the game to often Dispose & Allocate

File: internal/font/font.go (hunk 1 above)
Comment (wasedaigo) [reply internal/font/font.go]: yes, it makes it more readable.

Comment (hajimehoshi) [reply internal/font/font.go]: done

File: internal/font/font.go (hunk 2 above)
Comment (hajimehoshi) [reply internal/font/font.go]: This is technically not needed but as long as there is no reference, GC would call Dispose.

Allocating a new image is done on a shared texture, so there is (almost) no performance regression in the view of allocation and disposing.

Comment (hajimehoshi) [reply internal/font/font.go]: Oh, I realized that the rendering offscreen does not use a shared texture. Let me think.

Comment (hajimehoshi) [reply internal/font/font.go]: As allocating/disposing happens only when a bigger size is required, I think the performance can be  problematic not very likely.

**Pull request merged**